### PR TITLE
Bug fix: Add reached_reapplication_limit to non-editable path list

### PR DESCRIPTION
**Pull request by elceebee**
## Context
[Sentry Error](https://dfe-teacher-services.sentry.io/issues/5076372603/?referrer=slack&notification_uuid=971019e9-c997-4394-8c13-6f070096924a&alert_rule_id=853774&alert_type=issue)

### Steps to recreate

- Login as a candidate that has been rejected twice for the same course
- Create a new application choice,
- Chose a different course (same provider) and click continue
- Change the course to the original (already been rejected) course
- You’ll get an error

### Expected behaviour
GIVEN I am a candidate who has been rejected twice from a course
AND I create a new application choice with a *different* course from the *same* provider. 
WHEN I edit the course selection
AND I chose the original course, that I've already been rejected from
THEN I see the reapplication error page

## Changes proposed in this pull request

- Added `reached_reapplication_limit` to the list of non-editable paths. 
- Spec covering this scenarios

## Link to Trello card

[Trello Card](https://trello.com/c/yv6e8Zti/1439-course-already-applied-when-candidate-changes-their-course?filter=member:baileylori) 

## Things to check

- [x] If the code removes any existing feature flags, a data migration has also been added to delete the entry from the database
- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] If this code adds a column to the DB, decide whether it needs to be in analytics yml file or analytics blocklist
- [x] API release notes have been updated if necessary
- [x] If it adds a significant user-facing change, is it documented in the [CHANGELOG](CHANGELOG.md)?
- [x] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

**Review by tomas-stefano — APPROVED** [review]

File: spec/system/candidate_interface/continuous_applications/course_selection/candidate_changes_course_and_reaches_limit_spec.rb [hunk 1]
```diff
@@ -0,0 +1,84 @@
+require 'rails_helper'
+
+RSpec.feature 'Changing a course' do
+  include CandidateHelper
+
+  it 'Candidate changes course to one they have already been rejected from twice' do
+    given_i_am_signed_in
+    and_there_are_course_options
+    and_i_have_two_rejected_application_to_a_course
```
Comment (inulty-dfe) [inline spec/system/candidate_interface/continuous_applications/course_selection/candidate_changes_course_and_reaches_limit_spec.rb:9]: `applications`

File: app/forms/candidate_interface/continuous_applications/concerns/course_selection_step_helper.rb [hunk 2]
```diff
@@ -5,7 +5,7 @@ module CourseSelectionStepHelper
         delegate :application_choice, to: :store
 
         def next_edit_step_path(next_step_klass)
-          return next_step_path(next_step_klass) if next_step == :course_review
+          return next_step_path(next_step_klass) if %i[course_review reached_reapplication_limit].include?(next_step)
```
[the remark this answers is not in the document]
Comment (avinhurry) [reply app/forms/candidate_interface/continuous_applications/concerns/course_selection_step_helper.rb]: Updated here: https://github.com/DFE-Digital/apply-for-teacher-training/pull/9222/commits/ef2325b4469ba3f08f2e48b816c562f73d2b4e2e

File: spec/system/candidate_interface/continuous_applications/course_selection/candidate_changes_course_and_reaches_limit_spec.rb [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+    when_i_visit_the_site
```
[the remark this answers is not in the document]
Comment (avinhurry) [reply spec/system/candidate_interface/continuous_applications/course_selection/candidate_changes_course_and_reaches_limit_spec.rb]: Also updated here: https://github.com/DFE-Digital/apply-for-teacher-training/pull/9222/commits/ef2325b4469ba3f08f2e48b816c562f73d2b4e2e

File: spec/system/candidate_interface/continuous_applications/course_selection/candidate_changes_course_and_reaches_limit_spec.rb (hunk 1 above)
Comment (avinhurry) [reply spec/system/candidate_interface/continuous_applications/course_selection/candidate_changes_course_and_reaches_limit_spec.rb]: https://github.com/DFE-Digital/apply-for-teacher-training/pull/9222/commits/ef2325b4469ba3f08f2e48b816c562f73d2b4e2e

File: spec/system/candidate_interface/continuous_applications/course_selection/candidate_changes_course_and_reaches_limit_spec.rb (hunk 3 above)
[the remark this answers is not in the document]
Comment (inulty-dfe) [reply spec/system/candidate_interface/continuous_applications/course_selection/candidate_changes_course_and_reaches_limit_spec.rb]: Oh sorry, I didn't mention is explicitly but I think this method should be in the CandidateHelper.

**Pull request merged**